crates/vim/src/vim.rs

```diff
@@ -1014,9 +1012,11 @@ impl Vim {
             editor.set_input_enabled(vim.editor_input_enabled());
             editor.set_autoindent(vim.should_autoindent());
             editor.selections.line_mode = matches!(vim.mode, Mode::VisualLine);
-            if VimSettings::get_global(cx).smart_relative_line {
+            if VimSettings::get_global(cx).toggle_relative_line_numbers {
                 let is_relative = vim.mode != Mode::Insert;
-                editor.set_relativg_line_numbers(Some(is_relative), cx)
+                editor.set_relative_line_number(Some(is_relative), cx)
+            } else {
+                editor.set_relative_line_number(None, cx)
```

Comment:
If it's not enabled, using the default settings will cause temporary switching to fail, so we should preserve the original logic, which is to prioritize reading the configuration when it's not enabled, and then user manual toggle.